posts/2022-08-02-separate-stack-trace-22008.adoc

```diff
@@ -0,0 +1,466 @@
+                if (trace && tc.isEntryEnabled())
+                    Tr.exit(this, tc, "run[" + taskId + ']', "unavailable - deferred");
+                return; // Ignore, we are deferring the task because the application or module is unavailable
+            }
+----
++
+As presented in the code, the return statment from both of those if blocks. It prevents code flow from continuing, meaning that appTracker.deferTask is unreachable.
++
+This issue has now had a fix to run deferred tasks after the app starts in link:https://github.com/OpenLiberty/open-liberty/pull/21616[#21616] and has subsequently now been closed. 
+
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/21664[featureUpdate downloads fail in Windows, due to #20945]
++
+The issue is that you are unable to connect to Maven repo on Windows due to changes made in #20945. There is a need to append backsplash instead of File.separator for urls.
++
+The steps to reproduce this follows:
++
+[source, role="no_copy"]
+----
+featureUtility.exe installFeature or featureUtility.exe installServerFeature 
+..
+The remote repository "https://repo.maven.apache.org/maven2/\" is missing the following artifacts: [json-1.0-22.0.0.7.pom, json-1.0-22.0.0.7.esa]
+[7/27/22, 11:31:52:318 PDT] Downloading required features ...
+[7/27/22, 11:31:52:334 PDT] Starting installation ...
+[7/27/22, 11:31:52:365 PDT] Successfully completed cleaning up temporary files.
+<---------------------> Infinity%
+java.lang.NullPointerException
+at com.ibm.ws.install.featureUtility.FeatureUtility.installFeatures(FeatureUtility.java:483)
+at com.ibm.ws.install.featureUtility.cli.InstallFeatureAction.install(InstallFeatureAction.java:241)
+at com.ibm.ws.install.featureUtility.cli.InstallFeatureAction.execute(InstallFeatureAction.java:257)
+at com.ibm.ws.install.featureUtility.cli.InstallFeatureAction.handleTask(InstallFeatureAction.java:78)
+at com.ibm.ws.install.featureUtility.cli.FeatureAction.handleTask(FeatureAction.java:100)
+at com.ibm.ws.install.featureUtility.FeatureUtilityExecutor.main(FeatureUtilityExecutor.java:58)
+at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
+at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
+at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
+at java.base/java.lang.reflect.Method.invoke(Method.java:566)
+at com.ibm.ws.kernel.boot.cmdline.UtilityMain.internal_main(UtilityMain.java:173)
+at com.ibm.ws.kernel.boot.cmdline.UtilityMain.main(UtilityMain.java:53)
+at com.ibm.ws.kernel.boot.cmdline.Main.main(Main.java:52)
+----
++
+The expected behaviour is for the features to be installed successfully. 
++
+Diagnostic Information:
++
+OpenLiberty Version: [22.0.0.7]
++
+This issue has since been reviewed and link:https://github.com/OpenLiberty/open-liberty/pull/21667[#21667] reverts it back to a backsplash. This has now been closed. 
```

Comment:
```suggestion
* link:https://github.com/OpenLiberty/open-liberty/issues/21664[featureUpdate downloads fail in Windows, due to #20945]
+
The issue is caused by the inability to connect to Maven repo on Windows due to changes made in #20945. There is a need to append a forward slash (`/`) instead of `File.separator` for urls.
+
Running `featureUtility.exe installFeature` or`featureUtility.exe installServerFeature` causes the following:
+
[source, role="no_copy"]
----
The remote repository "https://repo.maven.apache.org/maven2/\" is missing the following artifacts: [json-1.0-22.0.0.7.pom, json-1.0-22.0.0.7.esa]
[7/27/22, 11:31:52:318 PDT] Downloading required features ...
[7/27/22, 11:31:52:334 PDT] Starting installation ...
[7/27/22, 11:31:52:365 PDT] Successfully completed cleaning up temporary files.
<---------------------> Infinity%
java.lang.NullPointerException
at com.ibm.ws.install.featureUtility.FeatureUtility.installFeatures(FeatureUtility.java:483)
at com.ibm.ws.install.featureUtility.cli.InstallFeatureAction.install(InstallFeatureAction.java:241)
at com.ibm.ws.install.featureUtility.cli.InstallFeatureAction.execute(InstallFeatureAction.java:257)
at com.ibm.ws.install.featureUtility.cli.InstallFeatureAction.handleTask(InstallFeatureAction.java:78)
at com.ibm.ws.install.featureUtility.cli.FeatureAction.handleTask(FeatureAction.java:100)
at com.ibm.ws.install.featureUtility.FeatureUtilityExecutor.main(FeatureUtilityExecutor.java:58)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at com.ibm.ws.kernel.boot.cmdline.UtilityMain.internal_main(UtilityMain.java:173)
at com.ibm.ws.kernel.boot.cmdline.UtilityMain.main(UtilityMain.java:53)
at com.ibm.ws.kernel.boot.cmdline.Main.main(Main.java:52)
----
+
This issue has since been resolved by link:https://github.com/OpenLiberty/open-liberty/pull/21667[#21667] which reverts the `File.separator` to a forward slash and the features now install successfully.
```